app/src/main/java/com/example/habitsmasher/ui/home/HomeFragment.java

```diff
@@ -1,66 +1,273 @@
+        recyclerView.addOnItemTouchListener(touchListener);
+    }
+
+    @Override
+    protected void openAddDialogBox() {
+        // Originally went to the dialog box but the box relies on the  list to function
+        //no longer possible
+
+    }
+
+    @Override
+    public void updateListAfterAdd(Object addedObject) {
+        // not needed
+    }
+
+    @Override
+    public void updateListAfterEdit(Object editedObject, int pos) {
+        //not needed
+    }
+
+    private Habit makeHabit(Map<String, Object> extractMap) {
+        // get all of the data from the snapshot
+        String title = (String) extractMap.get("title");
+        String reason = (String) extractMap.get("reason");
+        Timestamp date = (Timestamp) extractMap.get("date");
+        String id = (String) extractMap.get("id");
+        String days = extractMap.get("days").toString();
+        boolean isPublic = (boolean) extractMap.get("public");
+
+        return new Habit(title, reason, date.toDate(), days, isPublic, id, new HabitEventList());
+    }
+
+    private String getCurrentDay() {
+        LocalDate currentDate = LocalDate.now();
+        return currentDate.getDayOfWeek().toString().toUpperCase().substring(0, 2);
+    }
+
+    private String getDateString() {
+        LocalDate currentDate = LocalDate.now();
+        String abbreviatedMonth = currentDate.getMonth().toString().substring(0, 3).toLowerCase();
+        abbreviatedMonth = abbreviatedMonth.substring(0, 1).toUpperCase() + abbreviatedMonth.substring(1);
+        return abbreviatedMonth + " " + currentDate.getDayOfMonth();
+    }
+
+    // note: add this to list fragment class once view is implemented for habitevents
+    protected void openViewWindowForItem(int position) {
+        // Get the selected habit
+        Habit currentHabit = _habitItemAdapter._snapshots.get(position);
+
+        // Create a bundle to be passed into the habitViewFragment
+        Bundle bundle = new Bundle();
+        bundle.putSerializable("habit", currentHabit);
+        bundle.putSerializable("userId", _user.getId());
+        NavController controller = NavHostFragment.findNavController(this);
+
+        // Navigate to the habitViewFragment
+        controller.navigate(R.id.action_navigation_home_to_habitViewFragment, bundle);
+    }
+
+    /**
+     * Used to open up the list of habit events associated with this habit
+     */
+    private void openHabitEventsView(int position) {
+        // Create a bundle to be passed into the HabitEventListFragment
+        Habit currentHabit = _habitItemAdapter._snapshots.get(position);
+        Bundle bundle = new Bundle();
+        bundle.putSerializable("parentHabit", currentHabit);
+        bundle.putSerializable("parentUser", _user.getId());
+
+        NavController controller = NavHostFragment.findNavController(this);
+
+        // Navigate to the habitEventListFragment
+        controller.navigate(R.id.action_navigation_home_to_habitEventListFragment, bundle);
     }
```

Comment:
Both of these are essentially just copied from other functions with the new action IDs.